src/AutoRest.CSharp/Common/Generation/Writers/FormattableStringHelpers.cs

```diff
@@ -134,12 +134,22 @@ private static FormattableString GetConversionFromFrameworkToRequestContent(this
         {
             if (TypeFactory.IsReadWriteDictionary(parameter.Type))
             {
-                return $"{typeof(RequestContentHelper)}.{nameof(RequestContentHelper.FromDictionary)}({parameter.Name})";
+                var conversionString = $"{typeof(RequestContentHelper)}.{nameof(RequestContentHelper.FromDictionary)}({parameter.Name})";
+                if (parameter.IsOptionalInSignature)
```

Comment:
even if the parameter is not optional in signature, it still could be null.
Here we should check the validation of this parameter is not "AssertNotNull or AssertNotNullOrEmpty" or just check if the validation is "None"

Reply:
If the parameter is optional in signature, we will check null, so it cannot be null.

Reply:
well never mind - we could do that later maybe